### PR TITLE
lido_liquidity_arbitrum_wombat_pools

### PR DESCRIPTION
**Pull request by ppclunghe**
Dune SQL Test on wombat exchange wstETH pool for lido liquidity model

### CODE REVIEW

File: models/lido/liquidity/arbitrum/lido_liquidity_arbitrum_wombat_pools.sql [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  )
+
+, tokens_prices_daily AS (
+    SELECT DISTINCT
+      DATE_TRUNC('day', minute) AS time,
+      contract_address as token,
+      AVG(price) AS price
+    FROM {{source('prices','usd')}}
+    WHERE
+      DATE_TRUNC('day', minute) >= timestamp '{{ project_start_date }}'
```
Comment (Hosuke) [inline models/lido/liquidity/arbitrum/lido_liquidity_arbitrum_wombat_pools.sql:32]:
```suggestion
      DATE_TRUNC('day', minute) >= DATE '{{ project_start_date }}'
```

File: models/lido/liquidity/arbitrum/lido_liquidity_arbitrum_wombat_pools.sql [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+      LEAD(time,1,DATE_TRUNC('hour', now() + INTERVAL '1' hour)) OVER (ORDER BY time NULLS FIRST) AS next_time,
+      price
+    FROM
+      (
+    SELECT DISTINCT
+      DATE_TRUNC('hour', minute) AS time,
+      AVG(price) AS price
+    FROM {{source('prices','usd')}}
+    WHERE
+      DATE_TRUNC('day', minute) >= cast('{{ project_start_date }}' as timestamp)
```
Comment (Hosuke) [inline models/lido/liquidity/arbitrum/lido_liquidity_arbitrum_wombat_pools.sql:61]:
```suggestion
      DATE_TRUNC('day', minute) >= DATE '{{ project_start_date }}'
```

File: models/lido/liquidity/arbitrum/lido_liquidity_arbitrum_wombat_pools.sql [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+                                "project",
+                                "lido_liquidity",
+                                \'["ppclunghe"]\') }}'
+    )
+}}
+
+{% set project_start_date =  '2023-05-25'%} 
+
+with dates as (
+    with day_seq as (select (sequence(cast('{{ project_start_date }}' as date), cast(now() as date), interval '1' day)) as day)
```
Comment (Hosuke) [inline models/lido/liquidity/arbitrum/lido_liquidity_arbitrum_wombat_pools.sql:19]: Not sure if this CTE is valid, but you can test it on v2 dune sql first.

File: models/lido/liquidity/arbitrum/lido_liquidity_arbitrum_wombat_pools.sql [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    GROUP BY  1,2
+ )
+ 
+  , deposit_wsteth_events AS (
+    SELECT
+      DATE_TRUNC('day', sw.evt_block_time) AS time,
+      sw.contract_address AS pool,
+      SUM(cast(amount as double)) AS amount0
+    FROM {{source('wombat_arbitrum','wsteth_pool_evt_Deposit')}} AS sw
+    WHERE DATE_TRUNC('day', sw.evt_block_time) >= timestamp '{{ project_start_date }}'
```
Comment (Hosuke) [inline models/lido/liquidity/arbitrum/lido_liquidity_arbitrum_wombat_pools.sql:86]: Same as above, you may retry with all `timestamp` replaced by `DATE` first.
```suggestion
    WHERE DATE_TRUNC('day', sw.evt_block_time) >= DATE '{{ project_start_date }}'
```

File: models/lido/liquidity/arbitrum/lido_liquidity_arbitrum_wombat_pools.sql (hunk 1 above)
Comment (ppclunghe) [reply models/lido/liquidity/arbitrum/lido_liquidity_arbitrum_wombat_pools.sql]: thanks a lot, fixed and checked

**Comment by jeff-dude** [comment]
the PR title is a bit misleading -- what is the intention for this PR at this time? is this a new spell altogether and only want built on dunesql?

**Comment by ppclunghe** [comment]
no, it is the new sub-spell for lido liquidity spell
can we add  it to our spell or I should migrate whole spell previously?

**Comment by jeff-dude** [comment]
> no, it is the new sub-spell for lido liquidity spell can we add it to our spell or I should migrate whole spell with all sub-spells previously?

you will need to migrate all upstream spells first that roll-up into the full lido liquidity spell, we can't have cross-engine lineage. you're welcome to do one at a time in separate PRs, have them run standalone, then eventually once all migrated, can do the roll-up spell. keep in mind, with how the migration is running, old spark spells will continue to run and you'll have those as backup (and can query via dunesql on the app)

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**